crates/ruff/src/rules/tryceratops/rules/raise_within_try.rs

```diff
@@ -77,15 +82,55 @@ pub(crate) fn raise_within_try(checker: &mut Checker, body: &[Stmt], handlers: &
         return;
     }
 
+    // The names of exceptions handled by the `try` Stmt. We can't compare `Expr`'s since they have
+    // different ranges, but by virtue of this function's call path we know that the `raise`
+    // statements will always be within the surrounding `try`.
+    let handler_ids: FxHashSet<&str> = helpers::extract_handled_exceptions(handlers)
+        .into_iter()
+        .filter_map(|handler| {
+            if let Expr::Name(ast::ExprName { id, .. }) = handler {
+                Some(id.as_str())
+            } else {
+                None
+            }
+        })
+        .collect();
```

Comment:
Nit: It could be faster to use `Vec` here, considering we'll only have very few entries. We could also potentially make this lazy to avoid allocating the vector for try statements that have no `raise` statement.

Reply:
@MichaReiser Sorry - under what circumstances is a `Vec` faster than an `iter()`? Just for future reference

Reply:
Oh, sorry. Iterating over a `Vec` can be faster than looking up a set if you have very few items and if comparing the elements is cheap. This is mainly because sets have an additional level of indirection. It first needs to find the bucket, then compare the element in that bucket. We probably also don't bother about duplicates here. If we have a duplicate, so what...

Reply:
Interesting. I was using a set for the O(1) lookup. Is there a rule of thumb for when that `Vec` iteration is quicker?